src/lib.rs

```diff
@@ -17,87 +21,189 @@ impl Default for Syntax {
     }
 }
 
+/// A field frequency
 #[derive(Debug, Clone)]
 pub enum Frequency {
+    /// A well-formed message can have zero or one of this field (but not more than one).
     Optional,
+    /// This field can be repeated any number of times (including zero) in a well-formed message.
+    /// The order of the repeated values will be preserved.
     Repeated,
+    /// A well-formed message must have exactly one of this field.
```

Comment:
That's not exactly true. When protobuf parses message, last field in the stream wins. The same applies to optional fields.

Reply:
I am not sure I understand your point.

The documentation comes from [proto2 -Specifying Field Rules ](https://developers.google.com/protocol-buffers/docs/proto#specifying-field-rules).

Reply:
OK then!

(I meant when byte stream contains multiple fields with the same number, protobuf successfully parses it as optional or required using last value in the stream).